src/rpc/mining.cpp

```diff
@@ -887,72 +887,92 @@ UniValue estimaterawfee(const JSONRPCRequest& request)
-    passbucket.push_back(Pair("withintarget", round(buckets.pass.withinTarget * 100.0) / 100.0));
-    passbucket.push_back(Pair("totalconfirmed", round(buckets.pass.totalConfirmed * 100.0) / 100.0));
-    passbucket.push_back(Pair("inmempool", round(buckets.pass.inMempool * 100.0) / 100.0));
-    passbucket.push_back(Pair("leftmempool", round(buckets.pass.leftMempool * 100.0) / 100.0));
-    result.push_back(Pair("pass", passbucket));
-    UniValue failbucket(UniValue::VOBJ);
-    failbucket.push_back(Pair("startrange", round(buckets.fail.start)));
-    failbucket.push_back(Pair("endrange", round(buckets.fail.end)));
-    failbucket.push_back(Pair("withintarget", round(buckets.fail.withinTarget * 100.0) / 100.0));
-    failbucket.push_back(Pair("totalconfirmed", round(buckets.fail.totalConfirmed * 100.0) / 100.0));
-    failbucket.push_back(Pair("inmempool", round(buckets.fail.inMempool * 100.0) / 100.0));
-    failbucket.push_back(Pair("leftmempool", round(buckets.fail.leftMempool * 100.0) / 100.0));
-    result.push_back(Pair("fail", failbucket));
+    const char* horizonNames[] = {"short", "medium", "long"};
+
+    for (FeeEstimateHorizon horizon : {FeeEstimateHorizon::SHORT_HALFLIFE, FeeEstimateHorizon::MED_HALFLIFE, FeeEstimateHorizon::LONG_HALFLIFE}) {
+        CFeeRate feeRate;
+        EstimationResult buckets;
+        if ((unsigned int)nBlocks <= ::feeEstimator.HighestTargetTracked(horizon)) {
+            feeRate = ::feeEstimator.estimateRawFee(nBlocks, threshold, horizon, &buckets);
+            UniValue horizonresult(UniValue::VOBJ);
+            UniValue errors(UniValue::VARR);
+            UniValue passbucket(UniValue::VOBJ);
+            passbucket.push_back(Pair("startrange", round(buckets.pass.start)));
+            passbucket.push_back(Pair("endrange", round(buckets.pass.end)));
+            passbucket.push_back(Pair("withintarget", round(buckets.pass.withinTarget * 100.0) / 100.0));
+            passbucket.push_back(Pair("totalconfirmed", round(buckets.pass.totalConfirmed * 100.0) / 100.0));
+            passbucket.push_back(Pair("inmempool", round(buckets.pass.inMempool * 100.0) / 100.0));
+            passbucket.push_back(Pair("leftmempool", round(buckets.pass.leftMempool * 100.0) / 100.0));
+            UniValue failbucket(UniValue::VOBJ);
+            failbucket.push_back(Pair("startrange", round(buckets.fail.start)));
+            failbucket.push_back(Pair("endrange", round(buckets.fail.end)));
+            failbucket.push_back(Pair("withintarget", round(buckets.fail.withinTarget * 100.0) / 100.0));
+            failbucket.push_back(Pair("totalconfirmed", round(buckets.fail.totalConfirmed * 100.0) / 100.0));
+            failbucket.push_back(Pair("inmempool", round(buckets.fail.inMempool * 100.0) / 100.0));
+            failbucket.push_back(Pair("leftmempool", round(buckets.fail.leftMempool * 100.0) / 100.0));
+            if (!(feeRate == CFeeRate(0))) {
```

Comment:
I like having the success condition be the first thing you read, but I'll add a comment